### PR TITLE
Using new API for textStyles and colors exporting. 

### PR DESCRIPTION
**Pull request by acidlynx**
Also used styles from linked Styleguides.
This fixes issue https://github.com/amaurydavid/SwiftForAll/issues/1

### CODE REVIEW

File: src/swiftTextStyles.js [hunk 1]
```diff
@@ -7,8 +7,12 @@ var swiftColors = require('./swiftColors.js')
 
 function getTextStylesSwiftFileContent(context, textStyles) {
   var code = getSwiftFontImportCode(context) + "\n\n";
+  code += "//swiftlint:disable file_length\n";
+  code += "//swiftlint:disable identifier_name\n\n";
   code += getTextStylesSwiftSnippet(context,textStyles, true) + "\n\n";
   code += getSwiftConvenienceStringExt();
+  code += "\n\n//swiftlint:enable identifier_name\n";
+  code += "//swiftlint:enable file_length\n";
```
Comment (amaurydavid) [inline src/swiftTextStyles.js:15]: I would rather not include swiftlint comments by default as Swiftlint is not used by everyone and its usage greatly varies. Nonetheless, it could potentially be done in another PR where we would add a new option in the extension's settings.

File: src/index.js [hunk 2]
```diff
@@ -14,41 +14,54 @@ function layer(context, selectedLayer) {
...
     language: "swift"
   };
 }
 
-function styleguideTextStyles(context, textStyles) {
-  var code = "// MARK: - Project Texts styles\n\n";
-  code += swiftTextStyles.getTextStylesSwiftSnippet(context,textStyles, false)
+function exportTextStyles(context) {
+  var textStylesProject = context.project.textStyles
+  var textStylesAll = textStylesProject.concat(context.project.linkedStyleguide.textStyles)
```
Comment (amaurydavid) [inline src/index.js:33]: From the documentation I see 3 different ways to get textStyles:
- context.project.textStyles
- context.styleguide.textStyles
- context.project.linkedStyleguide.textStyles

You seem to use only 2 of them, any reason?

File: src/index.js [hunk 3]
```diff
@@ -14,41 +14,54 @@ function layer(context, selectedLayer) {
...
-function styleguideColors(context, colors) {
+function textStyles(context) {
+  var code = "// MARK: - Project Texts styles \n\n";
+  
+  var textStylesProject = context.project.textStyles
+  var textStylesAll = textStylesProject.concat(context.project.linkedStyleguide.textStyles)
 
-  var code = "// MARK: - Project Color palette\n\n";
-  code += swiftColors.getColorsSwiftSnippet(context, colors, false);
+  code += swiftTextStyles.getTextStylesSwiftSnippet(context, textStylesAll, false)
```
Comment (amaurydavid) [inline src/index.js:23]: Your code to fetch text styles (also works for colors) is duplicated for snippets and for export. Can you make a private function in swiftTextStyles.js to concat all textStyles together so that we no longer need to pass them in index.js ? It has the advantage of removing a parameter and keeping index.js at the minimum.

**Review by amaurydavid — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 2 above)
Comment (acidlynx) [reply src/index.js:33]: The second case is not working. Only from `project` and `project.linkedStyleguide`. But I'll check it. Thanks!

**Comment by amaurydavid** [comment]
I've added support for shared styleguides in 0.3.1.
Thanks @acidlynx for the heads-up and the work you made on this ;)

**Comment by acidlynx** [comment]
Welcome 👍
